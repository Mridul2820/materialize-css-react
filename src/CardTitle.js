import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import constants from './constants';

const CardTitle = ({ className, image, reveal, waves, children, ...props }) => {
  const classes = cx(className, {
    'card-image': true,
    'waves-effect': waves,
    'waves-block': waves,
    [`waves-${waves}`]: waves
  });

  return (
    <div className={cx(classes)} {...props}>
      <img className={cx({ activator: reveal })} src={image} />
      <span className={cx('card-title')}>{children}</span>
    </div>
  );
};

CardTitle.propTypes = {
  children: PropTypes.node,
  /**
   * Whether the image serves as activator for the reveal
   */
  reveal: PropTypes.bool,
  /**
   * the waves effect
   */
  waves: PropTypes.oneOf(constants.WAVES),
  /**
   * The path to the image
   */
  image: PropTypes.string.isRequired,
  /**
   * custom classname
   */
  className: PropTypes.string
};

export default CardTitle;
