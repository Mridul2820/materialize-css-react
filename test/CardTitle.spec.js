import React from 'react';
import { render } from '@testing-library/react';
import CardTitle from '../src/CardTitle';

describe('<CardTitle />', () => {
  test('renders', () => {
    const { container } = render(
      <CardTitle className="find" image="">
        I am a very simple card
      </CardTitle>
    );

    expect(container).toMatchSnapshot();
  });

  test('requires an image prop', () => {
    const { getByRole } = render(
      <CardTitle image="url.jpg" reveal>
        I am a very simple card
      </CardTitle>
    );

    expect(getByRole('img').className).toBe('activator');
  });

  test('can have waves', () => {
    const { container } = render(
      <CardTitle image="url.jpg" waves="green">
        I am a very simple card
      </CardTitle>
    );

    expect(container.firstChild.classList.contains('waves-green')).toBeTruthy();
  });
});
