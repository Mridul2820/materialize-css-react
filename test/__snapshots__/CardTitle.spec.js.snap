// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardTitle /> renders 1`] = `
<div>
  <div
    class="find card-image"
  >
    <img
      class=""
      src=""
    />
    <span
      class="card-title"
    >
      I am a very simple card
    </span>
  </div>
</div>
`;
